RoddyCore/src/de/dkfz/roddy/config/FilenamePattern.java

```diff
@@ -228,9 +232,21 @@ String fillConfigurationVariables(String src, ExecutionContext context) {
             // Only change it, if the value is in the configuration.
             if(configurationValues.hasValue(command.rawName)) {
                 src = src.replace(command.fullString, configurationValues.get(command.rawName).toString());
+            } else {
+                String value = "###" + blacklistID + "###";
```

Comment:
The purpose is to prevent the while loop to check for the same value again and again. If a variable is used but cannot be found in the config, it is blacklisted. In the next iteration it is not looked at.